wapitiCore/attack/mod_methods.py

```diff
@@ -39,6 +39,48 @@ class ModuleMethods(Attack):
     do_post = True
     excluded_path = set()
 
+    async def test_compare_methods(self, request: Request, methods: set):
+        response_dict = {}
+        # posting a GET request to compare
+        # with the uncommon methods
+        try:
+            log_verbose(f"[+] {request}")
+            get_response = await self.crawler.async_send(request)
+        except RequestError:
+            self.network_errors += 1
+            return
+        # Testing and comparing the uncommon methods,
+        # not_allowed_methods record all the methods
+        # returning a 405 error which will be removed
+        not_allowed_methods = set()
+        for method in methods:
+            uncommon_request = Request(
+                request.path,
+                method,
+                referer=request.referer,
+                link_depth=request.link_depth
+            )
+            try:
+                log_verbose(f"[+] {uncommon_request}")
+                response = await self.crawler.async_send(uncommon_request)
+            except RequestError:
+                self.network_errors += 1
+                return
+            if response.status == 405:
+                not_allowed_methods.add(method)
```

Comment:
Do you really need to have a second set? Maybe you could remove methods on the go, like

```suggestion
                methods - method
```

If you do it this way, you must modify line 82 to return _sorted(methods)_